core/src/gossip_service.rs

```diff
@@ -181,71 +181,55 @@ fn spy(
     let now = Instant::now();
     let mut met_criteria = false;
     let mut tvu_peers: Vec<ContactInfo> = Vec::new();
-    let mut archivers: Vec<ContactInfo> = Vec::new();
+    let mut storage_peers: Vec<ContactInfo> = Vec::new();
     let mut i = 0;
     loop {
         if let Some(secs) = timeout {
             if now.elapsed() >= Duration::from_secs(secs) {
                 break;
             }
         }
-        // collect tvu peers but filter out archivers since their tvu is transient and we do not want
-        // it to show up as a "node"
+
         tvu_peers = spy_ref
             .read()
             .unwrap()
             .all_tvu_peers()
             .into_iter()
             .collect::<Vec<_>>();
-        archivers = spy_ref.read().unwrap().all_storage_peers();
+        storage_peers = spy_ref.read().unwrap().all_storage_peers();
+
+        let mut nodes: Vec<_> = tvu_peers.iter().chain(storage_peers.iter()).collect();
+        nodes.sort();
+        nodes.dedup();
```

Comment:
Problem: `tvu_peers.len() + archivers.len() >= num` (aka, 1 node is not >= 2 nodes)

Reply:
Oops hehe 😅